File: 0649-dota2-senate/0649-dota2-senate.py
class Solution:
    def predictPartyVictory(self, senate: str) -> str:
        n = len(senate)
        radiant, dire = deque(), deque()                                                                             
        for i, char in enumerate(senate):
            if char == "R":
                radiant.append(i)
                
            else:
                dire.append(i)
                
        #when they are not empty
        while dire and radiant:
            rIndex = radiant.popleft()
            
            dIndex = dire.popleft()
           
            if rIndex < dIndex:
                radiant.append(rIndex+n)
                
                
            else:
                dire.append(dIndex+n)
                
        return 'Radiant' if radiant else 'Dire'
        

File: 0649-dota2-senate/0649-dota2-senate.scala
import scala.collection.mutable.Queue

object Solution {
    def predictPartyVictory(senate: String): String = {
        val n = senate.length
        val dir = Queue[Int]()
        val rad = Queue[Int]()
        for((char,i) <- senate.zipWithIndex){
            if (char == 'R') rad.enqueue(i)
            else dir.enqueue(i)
        }
        while(rad.nonEmpty && dir.nonEmpty){
            val rIndex = rad.dequeue()
            val dIndex = dir.dequeue()
            
            if (rIndex < dIndex) rad.enqueue(rIndex+n)
            else dir.enqueue(dIndex+n)
        }
        if (rad.nonEmpty) "Radiant"
        else "Dire"
    
        
    }
}